x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/collapsed_item_actions.tsx

```diff
@@ -51,28 +56,55 @@ export const CollapsedItemActions: React.FunctionComponent<ComponentOpts> = ({
   snoozeRule,
   unsnoozeRule,
 }: ComponentOpts) => {
-  const { ruleTypeRegistry } = useKibana().services;
+  const {
+    ruleTypeRegistry,
+    notifications: { toasts },
+  } = useKibana().services;
 
   const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
   const [isDisabled, setIsDisabled] = useState<boolean>(!item.enabled);
   useEffect(() => {
     setIsDisabled(!item.enabled);
   }, [item.enabled]);
 
+  const onClose = useCallback(() => {
+    setIsPopoverOpen(false);
+  }, [setIsPopoverOpen]);
+
   const snoozeRuleInternal = useCallback(
-    async (snoozeEndTime: string | -1, interval: string | null) => {
-      await snoozeRule(item, snoozeEndTime);
+    async (snoozeSchedule: SnoozeSchedule) => {
```

Comment:
originally I had these in the `rules_list_snooze_panel` component so we wouldn't have to duplicate the logic for `snoozeRuleInternal` and `unsnoozeRuleInternal`. But now it seems like we now have duplicated logic in `onApplySnooze` and `onApplyUnsnooze` in the notify badge component. 

I'll leave it up to you, but I think having the snooze and unsnooze logic together makes future maintenance easier

Reply:
Tracking this here: https://github.com/elastic/kibana/issues/134765

It's not necessarily a simple job to abstract away the `onClosePopover` and `onLoading` functions so that they work in multiple contexts where we're using the `SnoozePanel`, especially in the remaining spots where it's used in `RuleStatusDropdown`. I understand you worked around that with `RulesListSnoozePanel` but that ended up being a super complicated merge conflict to deal with. So I think similarly to the above comment, it would be much easier to deal with this as part of deprecating the Snooze option in `RuleStatusDropdown`.

We'd be doing that as part of https://github.com/elastic/kibana/issues/133803 I believe, because that's the last place we're including Snooze in the dropdown.